target/linux/realtek/dts-5.15/rtl9302_zyxel_xgs1250-12.dts

```diff
@@ -71,6 +72,107 @@
 							  // (5G, 10/100) (10G, 5G, 2.5G)
 		led_set2 = <0x0a20 0x0a01>; // LED set 2: 1000MBit, 10GBit
 	};
+
+	thermal-zones {
+		phy24-thermal {
+			/* Poll every 10 seconds */
+			polling-delay-passive = <10000>; /* milliseconds */
+			polling-delay = <10000>; /* milliseconds */
+			thermal-sensors = <&phy24>;
+
+			trips {
+				phy24_trip0: phy24-trip0 {
+					/* At 80 degrees turn on fan */
+					temperature = <80000>; /* milliCelsius */
+					hysteresis = <5000>; /* milliCelsius */
+					type = "active";
+				};
+
+				phy24_trip1: phy24-trip1 {
+					/* At 108 degrees phys exceed spec */
+					temperature = <108000>; /* milliCelsius */
+					hysteresis = <5000>; /* milliCelsius */
+					type = "critical";
+				};
+			};
+
+			cooling-maps {
+				map0 {
+					trip = <&phy24_trip0>;
+					cooling-device = <&chassis_fan THERMAL_NO_LIMIT THERMAL_NO_LIMIT>;
+				};
+			};
+		};
+
+		phy25-thermal {
+			/* Poll every 10 seconds */
+			polling-delay-passive = <10000>; /* milliseconds */
+			polling-delay = <10000>; /* milliseconds */
+			thermal-sensors = <&phy25>;
+
+			trips {
+				phy25_trip0: phy25-trip0 {
+					/* At 80 degrees turn on fan */
+					temperature = <80000>; /* milliCelsius */
+					hysteresis = <5000>; /* milliCelsius */
+					type = "active";
+				};
+
+				phy25_trip1: phy25-trip1 {
+					/* At 108 degrees phys exceed spec */
+					temperature = <108000>; /* milliCelsius */
+					hysteresis = <5000>; /* milliCelsius */
+					type = "critical";
+				};
+			};
+
+			cooling-maps {
+				map0 {
+					trip = <&phy25_trip0>;
+					cooling-device = <&chassis_fan THERMAL_NO_LIMIT THERMAL_NO_LIMIT>;
+				};
+			};
+		};
+
+		phy26-thermal {
+			/* Poll every 10 seconds */
+			polling-delay-passive = <10000>; /* milliseconds */
+			polling-delay = <10000>; /* milliseconds */
+			thermal-sensors = <&phy26>;
+
+			trips {
+				phy26_trip0: phy26-trip0 {
+					/* At 80 degrees turn on fan */
+					temperature = <80000>; /* milliCelsius */
+					hysteresis = <5000>; /* milliCelsius */
+					type = "active";
+				};
+
+				phy26_trip1: phy26-trip1 {
+					/* At 108 degrees phys exceed spec */
+					temperature = <108000>; /* milliCelsius */
+					hysteresis = <5000>; /* milliCelsius */
+					type = "critical";
+				};
+			};
+
+			cooling-maps {
+				map {
+					trip = <&phy26_trip0>;
+					cooling-device = <&chassis_fan THERMAL_NO_LIMIT THERMAL_NO_LIMIT>;
+				};
+			};
+		};
+	};
+
+	/* YEN SUN TECHNOLOGY FD122510LL-N fan */
+	chassis_fan: gpio-fan {
+		compatible = "gpio-fan";
+		gpios = <&gpio0 11 GPIO_ACTIVE_HIGH>;
```

Comment:
According to the board config, `<&gpio0 1>` would be "FAN CONTROL" (input). Any idea if that's an alarm/stall/... signal?

Reply:
I'm not entirely sure what it is. I would imagine it to be related to the fan tacho signal so it might be some type of stall indicator.

Reply:
Looking at the gpio value it seems this input is just sampling the tacho signal directly. So sadly nothing we could easily use as a failure indicator via standardized dts means I think.